test/object-store/sync/client_reset.cpp

```diff
@@ -4144,3 +4145,515 @@ TEST_CASE("client reset with embedded object", "[client reset][local][embedded o
+                auto obj = table->create_object_with_primary_key(pk_val);
+                auto col = table->get_column_key("any_mixed");
+                obj.set_collection(col, CollectionType::List);
+                List list{local, obj, col};
+                list.insert(0, Mixed{30});
+                REQUIRE(list.size() == 1);
+            })
+            ->make_remote_changes([&](SharedRealm remote_realm) {
+                auto table = get_table(*remote_realm, "TopLevel");
+                auto obj = table->create_object_with_primary_key(pk_val);
+                auto col = table->get_column_key("any_mixed");
+                obj.set_collection(col, CollectionType::Set);
+                object_store::Set set{remote_realm, obj, col};
+                set.insert(Mixed{40});
+                REQUIRE(set.size() == 1);
+            })
+            ->on_post_reset([&](SharedRealm local_realm) {
+                if (test_mode == ClientResyncMode::DiscardLocal)
+                    return;
+                advance_and_notify(*local_realm);
+                TableRef table = get_table(*local_realm, "TopLevel");
+                REQUIRE(table->size() == 1);
+                auto obj = table->get_object(0);
+                auto col = table->get_column_key("any_mixed");
+                List list{local_realm, obj, col};
+                REQUIRE(list.size() == 1);
+                REQUIRE(list.get_any(0) == 30);
+            })
+            ->run();
+    }
+    SECTION("add nested collection both locally and remotely List vs Dictionary") {
+        ObjectId pk_val = ObjectId::gen();
+        SyncTestFile config2(init_sync_manager.app(), "default");
+        config2.schema = config.schema;
+        auto test_reset = reset_utils::make_fake_local_client_reset(config, config2);
+        test_reset
+            ->make_local_changes([&](SharedRealm local) {
+                auto table = get_table(*local, "TopLevel");
+                auto obj = table->create_object_with_primary_key(pk_val);
+                auto col = table->get_column_key("any_mixed");
+                obj.set_collection(col, CollectionType::List);
+                List list{local, obj, col};
+                list.insert(0, Mixed{30});
+                REQUIRE(list.size() == 1);
+            })
+            ->make_remote_changes([&](SharedRealm remote_realm) {
+                auto table = get_table(*remote_realm, "TopLevel");
+                auto obj = table->create_object_with_primary_key(pk_val);
+                auto col = table->get_column_key("any_mixed");
+                obj.set_collection(col, CollectionType::Dictionary);
+                object_store::Dictionary dict{remote_realm, obj, col};
+                dict.insert("Test", Mixed{40});
+                REQUIRE(dict.size() == 1);
+            })
+            ->on_post_reset([&](SharedRealm local_realm) {
+                if (test_mode == ClientResyncMode::DiscardLocal)
```

Comment:
validate that the remote changes "won" for DiscardLocal mode

Reply:
This and other similar places in the tests should be updated before resolving.

Reply:
Perhaps you didn't understand; what I'm asking for here is that you expand this conditional to check for a different outcome in the local_realm when in discard local mode, like you do in the tests below. Don't just `return` and ignore this test for discard local mode, actually verify the results (same issue on line 4380).